libevmasm/AssemblyItem.h

```diff
@@ -170,8 +171,24 @@ class AssemblyItem
 	/// @returns true if the assembly item can be used in a functional context.
 	bool canBeFunctional() const;
 
-	void setLocation(langutil::SourceLocation const& _location) { m_location = _location; }
-	langutil::SourceLocation const& location() const { return m_location; }
+	void setLocation(langutil::SourceLocation const& _location) {
+		if (!m_debugData)
+			m_debugData = yul::DebugData::create(_location);
+		else
+			m_debugData->nativeLocation = _location;
+	}
+
+	langutil::SourceLocation const& location() const {
+		if (!m_debugData)
+		{
+			static langutil::SourceLocation defaultSourceLocation;
+			return defaultSourceLocation;
+		}
+		return m_debugData->nativeLocation;
+	}
+
+	void setDebugData(yul::DebugData::Ptr const& _debugData) { m_debugData = _debugData; }
+	yul::DebugData::Ptr debugData() const { return m_debugData; }
```

Comment:
@ekpyron  This is a real hack. At least for now I did not find an easy solution to do this in a clean way. 
This function should not be marked as `const`, if it returns data that can modify `m_debugData`.

Reply:
Somehow I initially did not like that we need to set the whole `debugData`again, even if only a part of the data changed. But maybe it's totally ok.